index.js

```diff
@@ -141,50 +138,52 @@ FacebookInsightStream.prototype._init = function ( callback ) {
 }
 
 FacebookInsightStream.prototype._initItem = function ( item ) {
+    var id = typeof item === 'string' ? item : item.id
```

Comment:
null will give you an object - just make sure theres no possibility for nulls in the input or take care of this end

Reply:
If somehow a page object arrives in the source as `NULL` there is not much we can do at this stage. I don't want to start filtering out nulls that should be page objects since as soon as we do that we can't consider the job as a success. The items themselves originate from Facebook so I don't see any way to recover from being passed nulls - it will simply fail as soon as we try to access the `id` attribute.